libraries/iio/iio.c

```diff
@@ -98,11 +113,107 @@ static struct iio_desc		*g_desc;
 /************************ Functions Definitions *******************************/
 /******************************************************************************/
 
+static inline int32_t _pop_sock(struct iio_desc *desc,
+				struct tcp_socket_desc **sock)
+{
+	return cb_read(desc->sockets, sock, sizeof(*sock));
+}
+
+static inline int32_t _push_sock(struct iio_desc *desc,
+				 struct tcp_socket_desc *sock)
+{
+	return cb_write(desc->sockets, &sock, sizeof(sock));
+}
+
+static inline int32_t _nb_active_sockets(struct iio_desc *desc)
+{
+	int32_t size;
+
+	cb_size(desc->sockets, &size);
+
+	return size / sizeof(struct tcp_socket_desc *);
+}
+
+/* Blocking until new socket available.
+ * Will iterate through a list of sockets */
+static int32_t _get_next_socket(struct iio_desc *desc)
+{
+	struct tcp_socket_desc	*sock;
+	int32_t			ret;
+	uint32_t		nb_active_sockets;
+
+	/* Get all new waiting sockets.
+	 * If none is available, wait until first connection */
+	do {
+		ret = socket_accept(desc->server, &sock);
+		if (ret == -EAGAIN) {
+			nb_active_sockets = _nb_active_sockets(desc);
+			if (nb_active_sockets == 0)
+				/* Wait until a connection exists */
+				continue;
```

Comment:
I would add a debouncing delay right before continue, something like `mdelay(1)`.

Reply:
I think is a good idea too.

Reply:
done